extension-base/src/main/java/com/azure/autorest/extension/base/plugin/JavaSettings.java

```diff
@@ -156,7 +156,8 @@ public static JavaSettings getInstance() {
                     "http-status-code-to-exception-type-mapping"),
                 getBooleanValue(host, "partial-update", false),
                 getBooleanValue(host, "custom-strongly-typed-header-deserialization", false),
-                getBooleanValue(host, "generic-response-type", false)
+                getBooleanValue(host, "generic-response-type", false),
+                getBooleanValue(host, "stream-style-serialization", false)
```

Comment:
For now I'd want to make this optional as this would be a major surprise to a library author when they updated the version of Autorest Java they're using. But once all shipping libraries are update this configuration should be removed and be the default, unmodifiable behavior.

Reply:
We might need more tests before the switch.

Currently codegen has option to generate tests with mocked data on serialization e.g. https://github.com/Azure/autorest.java/blob/main/vanilla-tests/src/test/java/fixtures/bodycomplex/generated/SawsharkTests.java

Before getting to mgmt, we probably want to do this for multiple RPs, as well as live tests.